src/Debug.h

```diff
@@ -22,16 +22,9 @@
 #if defined(assert)
 #undef assert
 #endif
-
-#if PURIFY
-#define assert(EX) ((void)0)
-#elif defined(NODEBUG)
-#define assert(EX) ((void)0)
-#elif STDC_HEADERS
-#define assert(EX)  ((EX)?((void)0):xassert( # EX , __FILE__, __LINE__))
-#else
-#define assert(EX)  ((EX)?((void)0):xassert("EX", __FILE__, __LINE__))
-#endif
+// re-do custom assert definitions
```

Comment:
This comment is stating why the direct include of a compat file is done. Especially given the undef of its safety wrapper.

Reply:
It's not clear to me, maybe you can reword it?

Reply:
IMO, the comment is not clear primarily because there are no (obvious) "custom definitions" above the comment -- it is not clear what "custom definitions" we are going to redo. Perhaps there is no shared understanding of what "custom" means in this context.

As always, this lack of clarity should be fixed by adjusting the source code. Explaining the source code comment in a GitHub comment is not sufficient if the source code comment is not clear. FWIW, the current "we force compat/assert.h include because we need to re-do custom assert definitions" [explanation](https://github.com/squid-cache/squid/pull/937/files#r759200066) does not explain why those definitions need to be redone.

Reply:
reverted in 67eafc2